src/linalg/statschol.jl

```diff
@@ -6,17 +6,21 @@ retains the original order unless singularity is detected.  Columns that are
 (computationally) linearly dependent on columns to their left are moved to the
 right hand side in a left circular shift.
 """
-function statscholesky(xtx::Symmetric{T}, tol::Real = -1) where {T<:AbstractFloat}
+function statscholesky(xtx::Symmetric{T}, tol::Real = 0.0) where {T<:AbstractFloat}
     n = size(xtx, 2)
-    chpiv = cholesky(xtx, Val(true), tol = T(-1), check = false)
+    chpiv = cholesky(xtx, Val(true), tol = T(tol), check = false)
     chunp = cholesky(xtx, check = false)
-    r = chpiv.rank
+    r = rank(xtx) #chpiv.rank
```

Comment:
No, not yet discussed and one of the things I'm worried about. I had originally introduced it as a debugging step and my preference would be to avoid using expensive computations here (even though this is only run once to construct the model) so that we can have very large fixed-effects matrices. There was some discussion on using a different decomposition to avoid any latent numerical stability issues with the Cholesky factorization.